satellite/overlay/service.go

```diff
@@ -436,6 +440,12 @@ func (service *Service) GetMissingPieces(ctx context.Context, pieces []*pb.Remot
 	return missingPieces, nil
 }
 
+// GetNodeIPs returns a list of node IP addresses.
+func (service *Service) GetNodeIPs(ctx context.Context, nodeIDs storj.NodeIDList) (nodeIPs []string, err error) {
```

Comment:
Currently the reponsibility for properly selecting nodes for upload is in satellite/overlay, so the logic for handling that should be in overlay rather than anywhere else. The `GetNodeIPs` makes it look like that it will be added to somewhere else instead.

Reply:
I'm not sure I totally understand the problem. What would be a better way to organize the logic so that it makes sense?